collector/monitoring.go

```diff
@@ -0,0 +1,461 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// MonitoringCollector collects metrics about all droplets.
+type MonitoringCollector struct {
+	logger  log.Logger
+	errors  *prometheus.CounterVec
+	client  *godo.Client
+	timeout time.Duration
+
+	CPUMetrics             *prometheus.Desc
+	MemoryTotalMetrics     *prometheus.Desc
+	MemoryFreeMetrics      *prometheus.Desc
+	MemoryAvailableMetrics *prometheus.Desc
+	MemoryCachedMetrics    *prometheus.Desc
+	FileSystemFreeMetrics  *prometheus.Desc
+	FileSystemSizeMetrics  *prometheus.Desc
+	BandwidthMetrics       *prometheus.Desc
+}
+
+// NewMonitoringCollector returns a new DropletCollector.
+func NewMonitoringCollector(logger log.Logger, errors *prometheus.CounterVec, client *godo.Client, timeout time.Duration) *MonitoringCollector {
+	errors.WithLabelValues("droplet").Add(0)
+
+	labels := []string{"id", "name", "region"}
+	return &MonitoringCollector{
+		logger:  logger,
+		errors:  errors,
+		client:  client,
+		timeout: timeout,
+
+		CPUMetrics: prometheus.NewDesc(
+			"digitalocean_monitoring_cpu",
```

Comment:
Going with the other Prometheus metrics in the process Exporter this should probably have a _seconds_total suffix.